app/src/main/java/org/oppia/android/app/policies/PoliciesFragmentPresenter.kt

```diff
@@ -49,35 +60,64 @@ class PoliciesFragmentPresenter @Inject constructor(
       policyDescription =
         resourceHandler.getStringInLocale(R.string.privacy_policy_content)
       policyWebLink = resourceHandler.getStringInLocale(R.string.privacy_policy_web_link)
+      binding.policyDescriptionTextView.text = htmlParserFactory.create(
+        policyOppiaTagActionListener = this,
+        displayLocale = resourceHandler.getDisplayLocale()
+      ).parseOppiaHtml(
+        policyDescription,
+        binding.policyDescriptionTextView,
+        supportsLinks = true,
+        supportsConceptCards = false
+      )
+
+      binding.policyWebLinkTextView.text = htmlParserFactory.create(
+        gcsResourceName = "",
+        entityType = "",
+        entityId = "",
+        imageCenterAlign = false,
+        customOppiaTagActionListener = null,
+        resourceHandler.getDisplayLocale()
+      ).parseOppiaHtml(
+        policyWebLink,
+        binding.policyWebLinkTextView,
+        supportsLinks = true,
+        supportsConceptCards = false
+      )
     } else if (policyPage == PolicyPage.TERMS_OF_SERVICE) {
       policyDescription =
         resourceHandler.getStringInLocale(R.string.terms_of_service_content)
-      policyWebLink = resourceHandler.getStringInLocale(R.string.terms_of_service_web_link)
-    }
-
-    binding.policyDescriptionTextView.text = htmlParserFactory.create(
-      policyOppiaTagActionListener = this,
-      displayLocale = resourceHandler.getDisplayLocale()
-    ).parseOppiaHtml(
-      policyDescription,
-      binding.policyDescriptionTextView,
-      supportsLinks = true,
-      supportsConceptCards = false
-    )
+      val spannableString = SpannableString(HtmlCompat.fromHtml(policyDescription, HtmlCompat.FROM_HTML_MODE_LEGACY))
+      val uri = "https://www.oppia.org"
+      val spanStart = policyDescription.indexOf(uri) - 4
```

Comment:
Why are re using a fixed value of 4 here? What does it represent?

Reply:
the spanStart was returning the index of the fifth character in the link i.e 's' from https://www.oppia.org. Only fix was to subtract 4 from the starting index as i couldn't figure out why.

Reply:
This seems brittle, what happens if the uri were to change? Can you look for an inbuildt function for extracting the first part of a uri? You could also use pattern matching/regex.

Reply:
Hey @adhiamboperes  I have fixed and upated the PR. Please check.